pkg/proc/proc_linux.go

```diff
@@ -0,0 +1,85 @@
+package proc
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"k8s.io/klog"
+)
+
+// parseProcForZombies parses the current procfs mounted at /proc
+// to find processes in the zombie state.
+func parseProcForZombies() ([]int, error) {
+	files, err := ioutil.ReadDir("/proc")
+	if err != nil {
+		return nil, err
+	}
+
+	var zombies []int
+	for _, file := range files {
+		processID, err := strconv.Atoi(file.Name())
+		if err != nil {
+			break
+		}
+		stateFilePath := filepath.Join("/proc", file.Name(), "status")
+		fd, err := os.Open(stateFilePath)
+		if err != nil {
+			klog.Errorf("Failed to open %q: %v", stateFilePath, err)
+			continue
+		}
+		defer fd.Close()
+		fs := bufio.NewScanner(fd)
+		for fs.Scan() {
+			line := fs.Text()
+			if strings.HasPrefix(line, "State:") {
+				if strings.Contains(line, "zombie") {
+					zombies = append(zombies, processID)
+				}
+				break
+			}
+		}
+	}
+
+	return zombies, nil
+}
+
+// StartReaper starts a goroutine to reap processes periodically if called
+// from a pid 1 process.
+// If period is less than 5 seconds, then it is set to 5 seconds.
+// A caller can adjust the period depending on how many and how frequently zombie
+// processes are created and need to reaped. A process generating lots of zombie
+// processes should set the period to 5 seconds. A process that has fewer zombie
+// processes and can tolerate zombies being around longer could set it to a higher
+// value.
+func StartReaper(period time.Duration) {
+	if os.Getpid() == 1 {
+		const minReaperPeriodSeconds = 5
+		if period.Seconds() < minReaperPeriodSeconds {
+			period = minReaperPeriodSeconds * time.Second
+		}
+		go func() {
+			var zs []int
+			var err error
+			for {
+				zs, err = parseProcForZombies()
+				if err != nil {
+					klog.Errorf("Reaper: failed to parse zombies: %v", err)
```

Comment:
`failed to parse proc filesystem to find processes to terminate`

klog will already set the filename so `reaper: ` is unnecessary